File: Linked_List/Singly_Linked_List/ex4_7.java
// Inserting Node

package Singly_Linked_List;

public class ex4_7 {
    public static void main(String[] arg) {
        Node fourth = new Node(10);
        Node third = new Node(9, fourth); 
        Node second = new Node(8,third);
        Node first = new Node(7,second);

        // Insert x between second and third
        Node x = new Node(); // item of x = null, next of x = null
        x.setItem(8.5); // set item of x = 8.5
        x.setNext(third); // nextNode of x is third

        second.setNext(x); // nextNode of second is x

        Node curr = new Node();
        Node prev = new Node();
        curr = first;

        for(int i=0; i<4; i++) {
            if(curr.getItem().equals(8.5)) {
                System.out.println(curr.getNext());
                System.out.println(prev.getNext());
            }
            else {
                prev = curr;
                curr = curr.getNext();
            }
        }
    }
}

File: Linked_List/Singly_Linked_List/ex4_7.c
#include "C:\\Users\\kritt\\Documents\\Personal_Project\\Data_Structure_and_Algorithm\\Linked_List\\Singly_Linked_List\\Node.h"

int main() {
    struct Node *fourth = createNode(10, NULL);
    struct Node *third = createNode(9, NULL);
    struct Node *second = createNode(8, third);
    struct Node *first = createNode(7, second);

    setNext(third, fourth);

    struct Node *x;
    struct Node *curr;
    struct Node *prev;

    x = createNode(50, NULL);

    setNext(x, third);
    setNext(second, x);

    curr = first;
    prev = NULL;

    for(int i=0; i<10; i++) {
        if(getItem(curr) == 50) {
            println("%d", getItem(curr));
        }
        else {
            prev = curr;
            curr = getNext(curr);
        }
    }

    return 0;
}